src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/FieldBlocks.js

```diff
@@ -31,6 +71,38 @@ export default class FieldBlocks extends React.Component<FieldTypeProps<Array<Bl
         onFinish();
     };
 
+    getBlockSchemaType = (type: ?string) => {
+        const {defaultType, types, schemaPath} = this.props;
+
+        if (!type) {
+            throw new Error(
+                'It is impossible that a block has no type. This should not happen and is likely a bug.'
+            );
+        }
+
+        if (!types) {
+            throw new Error(MISSING_BLOCK_ERROR_MESSAGE);
+        }
+
+        if (types[type]) {
+            return types[type];
+        }
+
+        if (!defaultType) {
+            throw new Error(
+                'It is impossible that a block has no defaultType. This should not happen and is likely a bug.'
+            );
+        }
+
+        if (!types[defaultType]) {
+            throw new Error(
+                'The default type should exist in block "' + schemaPath + '".'
+            );
+        }
+
+        return types[defaultType];
```

Comment:
Do we still need the default type handling here? Is that because it will be rendered once before `componentDidUpdate` is called?

Reply:
@danrot yes we still need this here it will fail without it.